src/lib/math/bigint/bigint.h

```diff
@@ -665,26 +747,36 @@ class BOTAN_PUBLIC_API(2, 0) BigInt final {
        * @param buf destination byte array for the integer value
        * @param len how many bytes to write
        */
-      void binary_encode(uint8_t buf[], size_t len) const;
+      BOTAN_DEPRECATED("Use BigInt::serialize_to") void binary_encode(uint8_t buf[], size_t len) const;
+
+      /**
+       * Read integer value from a byte vector (big endian)
+       * @param bytes the span of bytes to load
+       */
+      void assign_from_bytes(std::span<const uint8_t> bytes);
```

Comment:
Do we even need that? Or is a simple assignment enough?

```C++
BigInt some_existing_bn;
some_existing_bn = BigInt::from_bytes(some_byte_array);
```

... with an appropriate move-assignment operator that might not even have a performance penalty.

Reply:
See also: https://github.com/randombit/botan/pull/4056#discussion_r1615775185